reconstructSPI/iterative_refinement/expectation_maximization.py

```diff
@@ -206,25 +222,29 @@ def iterative_refinement(self, wiener_small_number=0.01, count_norm_const=1):
 
                 for one_slice_idx in range(len(bayes_factors_1)):
                     xyz = xyz_rotated[one_slice_idx]
-                    inserted_slice_3d_r, count_3d_r = IterativeRefinement.insert_slice(
-                        particle_f_deconv_1.real, xyz, n_pix
+                    inserted_slice_3d_r, count_3d_r = insert_slice_v(
+                        particle_f_deconv_1.real, xyz, xyz_voxels
+                    )
+                    inserted_slice_3d_i, count_3d_i = insert_slice_v(
+                        particle_f_deconv_1.imag, xyz, xyz_voxels
                     )
-                    inserted_slice_3d_i, count_3d_i = IterativeRefinement.insert_slice(
-                        particle_f_deconv_1.imag, xyz, n_pix
+                    map_3d_f_updated_1 += np.sum(
+                        inserted_slice_3d_r + 1j * inserted_slice_3d_i, axis=0
                     )
-                    map_3d_f_updated_1 += inserted_slice_3d_r + 1j * inserted_slice_3d_i
-                    counts_3d_updated_1 += count_3d_r + count_3d_i
+                    counts_3d_updated_1 += np.sum(count_3d_r + count_3d_i, axis=0)
 
                 for one_slice_idx in range(len(bayes_factors_2)):
                     xyz = xyz_rotated[one_slice_idx]
-                    inserted_slice_3d_r, count_3d_r = IterativeRefinement.insert_slice(
-                        particle_f_deconv_2.real, xyz, n_pix
+                    inserted_slice_3d_r, count_3d_r = insert_slice_v(
+                        particle_f_deconv_2.real, xyz, xyz_voxels
                     )
-                    inserted_slice_3d_i, count_3d_i = IterativeRefinement.insert_slice(
-                        particle_f_deconv_2.imag, xyz, n_pix
+                    inserted_slice_3d_i, count_3d_i = insert_slice_v(
+                        particle_f_deconv_2.imag, xyz, xyz_voxels
                     )
-                    map_3d_f_updated_2 += inserted_slice_3d_r + 1j * inserted_slice_3d_i
-                    counts_3d_updated_2 += count_3d_r + count_3d_i
+                    map_3d_f_updated_2 += np.sum(
+                        inserted_slice_3d_r + 1j * inserted_slice_3d_i, axis=0
+                    )
+                    counts_3d_updated_2 += np.sum(count_3d_r + count_3d_i, axis=0)
```

Comment:
what's the return shape on `insert_slice_v`? Modularize things and write some tests and docs so that's clear.

what is being summed over in `np.sum(..., axis=0)`?

Reply:
insert_slice_v returns vector versions of the outputs of insert_slice, i.e., np.array(counts_3d_0, counts_3d_1, ...). 

The sum is adding together all of the slices that were inserted by this call of insert_slice_v. 

I don't know of a great way to modularize this as np.vectorize is the only way to generate the vectorized function. Open to suggestions.